cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -327,7 +327,29 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
         }
 
         Future<ReconciliationState> zkAncillaryCm() {
-            return withZkAncillaryCmChanged(configMapOperations.reconcile(namespace, zkCluster.getAncillaryConfigName(), zkMetricsAndLogsConfigMap));
+            Future<ReconciliationState> result = Future.future();
+
+            vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
+                future -> {
+                    ConfigMap current = configMapOperations.get(namespace, zkCluster.getAncillaryConfigName());
+                    boolean onlyMetricsSettingChanged = onlyMetricsSettingChanged(current, zkMetricsAndLogsConfigMap);
+                    future.complete(onlyMetricsSettingChanged);
+                }, res -> {
+                    if (res.succeeded())  {
+                        boolean onlyMetricsSettingChanged = (boolean) res.result();
```

Comment:
This typecast is unnecessary if you provide type argument `vertx.createSharedWorkerExecutor("kubernetes-ops-pool").<Boolean>executeBlocking(...)`